src/App.js

```diff
@@ -11,17 +11,33 @@ import LoginSuccess from "./pages/login-success/LoginSuccess";
 function App() {
   const location = useLocation();
 
+  const initializeAppForLoggedInUser = async (token) => {
+    const response = await fetch('https://api.spotify.com/v1/me',{
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    });
+
+    if (response.ok) {
+      const json = await response.json();
+
+      rootStore.userStore.setUserResponseData(json);
+    } else {
+      // TODO: make sure to do proper error handling
+    }
+  }
+
   // When the app loads, and we detect a token that is saved in localStorage,
   // we need to make sure that the userStore knows that the user is logged in.
   //
   // This will make sure that certain components doesn't show in the app
   // when the user is logged in (i.e. the login button and the sign up button)
-
   useEffect(() => {
     const token = localStorage.getItem("token");
 
     if (token) {
       rootStore.userStore.setIsLoggedIn(true);
+      initializeAppForLoggedInUser(token);
```

Comment:
Step 5: We need to call the function we created in step 4 inside of this `useEffect` hook